test-case-reporting/app.ts

```diff
@@ -47,8 +57,8 @@ function handleError(error: Error, res: express.Response): void {
 function extractPageInfo(req: express.Request): PageInfo {
   const {limit, offset} = req.query;
 
-  let limitNum = parseInt(limit.toString(), 10);
-  const offsetNum = parseInt(offset.toString(), 10);
+  let limitNum = limit ? parseInt(limit.toString(), 10) : DEFAULT_PAGE_SIZE;
```

Comment:
```suggestion
  const limitNum = limit 
    ? Math.min(parseInt(limit.toString()), MAX_PAGE_SIZE)
    : DEFAULT_PAGE_SIZE;
```
Nit: since the warning about the query size is only in the backend logs and not visible to the client/request, and since we probably won't care much about it on the backend, I'd just skip the logging entirely. Also, I know it used to be the case that parseInt required the second parameter to satisfy linters, but I have a vague memory of it no longer being required (maybe that's from another repo's conventions though, see what the linter says)

Reply:
I could imagine someone getting confused at why they're only getting 500 results -- is there a reasonable way to send the warning to the client/request? Do you believe it would be worth it?

You're right about the radix; I'll remove it.

Reply:
I'll apply the suggested changes, and if we decide to report the page size restriction later, we can make an issue.

Reply:
Not really, since the UI is likely to have a "1-500 of 10000" or something, and any pagination buttons would link to a url with the "correct" params, so upon clicking a link it'll be set properly anyway. TL;DR I don't think it's a common scenario and I don't think anyone would be too surprised to see a page limit in place, and it's a more graceful scenario than the alternative, which is responding with an error. If we wanted to nitpick we could send a redirect to the url with the adjusted parameter value(s) or embed an error/warning message in the page indicating the results were truncated, but I'd put those wayyyyyy down on the list of priorities under the "Does not merit the developer time required" heading :laughing: